First version

trocolit2 opened the pull request:
Please check my code!

trocolit2 committed:
Removed initial dummy files

trocolit2 committed:
Added source files and CMakeLists definitions

trocolit2 committed:
Added tests files and CMakeLists definitions

trocolit2 committed:
Added data set of image and video

trocolit2 committed:
Indent correction

romulogcerqueira commented on src/ImageEnhanceViaFusion.cpp:
```diff
@@ -0,0 +1,118 @@
+namespace underwater_image_tools {
+
+
+// percent should have values between 0.0001 and 0.05
```
This method should contain a check for `percent_channel` value

romulogcerqueira commented on src/ImageEnhanceViaFusion.cpp:
```diff
@@ -0,0 +1,118 @@
+
+
+// percent should have values between 0.0001 and 0.05
+cv::Mat colorBalance( cv::Mat image, const cv::Scalar& percent_channel ){
```
Once you are passing, modifying and returning the same `image`, you can pass this `cv::Mat` as reference and avoid copies.

romulogcerqueira commented on test/ImageEnhanceViaFusion_test.cpp:
```diff
@@ -0,0 +1,93 @@
+#include <underwater_image_tools/ImageEnhanceViaFusion.hpp>
+
+
+#define PATH_DATASET "../test/dataset/"
```
Try to use this as absolute path, for example:

<pre>
std::string current_path(__FILE__);
current_path = current_path.substr(0, current_path.find_last_of("/"));
</pre>

trocolit2 committed:
Refactored code, and followed the suggestions:

Captured the data set absolute path;
Return no image result after processing
Passed image as reference;

trocolit2 commented:
Code changed. Please review!

trocolit2 committed:
Removed the image channels check

trocolit2 committed:
Fixed path_dataset spelling
Added gray image test

trocolit2 committed:
Changed files name and updated CMakeLists files

romulogcerqueira commented:
It's good to me. Merging here.
